core/src/test/java/io/questdb/test/griffin/SqlOptimiserTest.java

```diff
@@ -0,0 +1,182 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin;
+
+import io.questdb.griffin.SqlCompiler;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.model.ExecutionModel;
+import io.questdb.griffin.model.ExpressionNode;
+import io.questdb.griffin.model.QueryModel;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayDeque;
+
+import static io.questdb.griffin.SqlOptimiser.aliasAppearsInFuncArgs;
+
+public class SqlOptimiserTest extends AbstractSqlParserTest {
+
+    @Test
+    public void testAliasAppearsInFuncArgs1() throws Exception {
+        assertMemoryLeak(() -> {
+            ddl("create table y ( x int );");
+            final String query = "select x1, sum(x1) from (select x x1 from y)";
+            final QueryModel model = compileModel(query, ExecutionModel.QUERY);
+            TestUtils.assertEquals("select-group-by x1, sum(x1) sum from (select-choose [x x1] x x1 from (select [x] from y))", model.toString0());
+            ArrayDeque<ExpressionNode> sqlNodeStack = new ArrayDeque<ExpressionNode>();
+            assert aliasAppearsInFuncArgs(model, "x1", sqlNodeStack);
+            assertPlan(query,
+                    "" +
```

Comment:
nit: do we need this `"" +` line here and below?

Reply:
In general, we prefer the following convention (yet, not all code follows the convention):
```java
            assertPlan(
                    query,
                    "GroupBy vectorized: true workers: 1\n" +
                            "  keys: [x1]\n" +
                            "  values: [sum(x1)]\n" +
                            "    SelectedRecord\n" +
                            "        DataFrame\n" +
                            "            Row forward scan\n" +
                            "            Frame forward scan on: y\n"
            );
```

Reply:
Since it's a nit comment, it's up to you whether to address it or not.